components/browser/session/src/main/java/mozilla/components/browser/session/storage/BrowserStateSerializer.kt

```diff
@@ -37,7 +37,13 @@ class BrowserStateSerializer {
         return json.toString()
     }
 
-    private fun tabToJSON(tab: TabSessionState): JSONObject {
+    /**
+     * Serializes the provided [TabSessionState] to JSON.
+     *
+     * @param tab the [TabSessionState] to serialize.
+     * @return the serialized state.
+     */
+    fun tabToJSON(tab: TabSessionState): JSONObject {
```

Comment:
If we only save the `EngineSessionState` to disk then we do not need to open this up here. `EngineSessionState` has `toJSON()` and you can reverse it via `Engine.createSessionState(json)`.